### DISCUSSION TITLE
Can't publish on quarto-pub

### DISCUSSION THREAD

**Discussion by pnavaro** [quarto-dev/quarto-cli#3940] [Q&A] [answered]
I am trying to publish on quarto-pub and I am stuck in a loop hole. I always get this message

```
The authorization saved for Quarto Pub is no longer valid.
  Please be sure you are logged into the correct Quarto Pub account in your
  default web browser, then press Enter to re-authorize.
[✓] Creating quarto-pub site
? Re-authorize account (Y/n) › Yes
? Authorize (Y/n) › Yes
```

I try a lot of things:
- Delete my account on quarto-pub and recreate (I used github and I change to email+passwd)
- Remove my account with `quarto publish accounts`
- Remove the directory `.quarto`
- Change my default browser (Safari, Firefox, Chrome)
- create a token and set the env variable `QUARTO_PUB_AUTH_TOKEN`

Thanks for your help

Pierre

**Comment by stevecrawshaw** [comment]
yes, me too..

**Comment by salimkaplan** [comment]
I have the same. It keeps asking for authorization

**Comment by cderv** [comment]
Thanks for the report. I have open an issue to track resolution: https://github.com/quarto-dev/quarto-cli/issues/3941

No need to everyone comment. You can add 👍 to confirm but we are aware of the issue. Thanks

**Comment by softwarenerd** [comment, marked as answer]
Hi All,

We identified a configuration issue that was causing publishing to fail with an authorization error. The configuration is now fixed, and publishing is back online. We understand the root cause of the issue, so this won't happen again.

Apologies for the inconvenience!

Best,

Brian

**Comment by higgi13425** [comment]
Looks like a version of this is happening again in February 2024. I have a daily cron job which I set up after an initial manual authorization. After an unclear period, it "loses" this authentication and the cron job stops running (recently on Jan 29, 2024, again on Feb 14, 2024). Is there a way to make this authentication more durable? 2 weeks seems a bit short for an automated process. Couldn't this be set up for annual, or q6 months? Kind of defeating the point of automation with a 2 week re-auth.

**Reply by cderv** [reply]
Just to be sure and cover everything before re-opening a configuration problem 

Did you set up Publishing credential in your cron-job using `QUARTO_PUB_AUTH_TOKEN` ? 
See our doc: 
- https://quarto.org/docs/publishing/quarto-pub.html#access-tokens
- https://quarto.org/docs/publishing/ci.html#publishing-credentials

If you are using the interactive way without setting up the auth token, then there is a re-auth delay by security. This is usual practice AFAIK. When doing automation using API TOKEN is recommended. 

or does it happen also with `QUARTO_PUB_AUTH_TOKEN` set ? In that case, this is definitely something we'll have a look in quarto pub configurations.

**Comment by higgi13425** [comment]
QUARTO_PUB_AUTH_TOKEN="hash"  is set at the top of the cron file, but it does not include 'export'.
I did manually run 'export QUARTO_PUB_AUTH_TOKEN="hash" ' some months ago, and it was working daily 
until it stopped working on Jan 29.
Is it best practice to export QUARTO_PUB_AUTH_TOKEN="hash" every time the cron job is run?
If so, it would be helpful to state that in the docs.

**Reply by cderv** [reply]
> Is it best practice to export QUARTO_PUB_AUTH_TOKEN="hash" every time the cron job is run?

I don't know exactly how cronjob script works and sets environment variables. I do know that on Linux usually you need to `export` a variable so that it is taken into account as an environment variable by the process that will follow. Otherwise, it is just a bash variable. 

**And so `quarto publish` does expect `QUARTO_PUB_AUTH_TOKEN` to be an environment.** So you need to make sure the environment variable is set for `quarto` to read it and find the access token.

> If so, it would be helpful to state that in the docs.

We do show example about how to use `export` on linux to set env var before calling `quarto publish`. We just don't have an example specific to cron job - as of now we don't plan to document cronjob in our documentation. 

> Is there a way to make this authentication more durable? 2 weeks seems a bit short for an automated process.

I asked team behind quartopub service about authentication delay when using access token, if any as we interactive credential flow. I'll report what they say when I have an answer.

**Comment by higgi13425** [comment]
Thanks - building an export token line into the start of the cron job works very well.Hindsight - Just wish I had done that from the beginning.Peter
    On Monday, February 17, 2025 at 09:42:00 AM EST, Christophe Dervieux ***@***.***> wrote:  
 
 



Is it best practice to export QUARTO_PUB_AUTH_TOKEN="hash" every time the cron job is run?


I don't know exactly how cronjob script works and sets environment variables. I do know that on Linux usually you need to export a variable so that it is taken into account as an environment variable by the process that will follow. Otherwise, it is just a bash variable.

And so quarto publish does expect QUARTO_PUB_AUTH_TOKEN to be an environment. So you need to make sure the environment variable is set for quarto to read it and find the access token.


If so, it would be helpful to state that in the docs.


We do show example about how to use export on linux to set env var before calling quarto publish. We just don't have an example specific to cron job - as of now we don't plan to document cronjob in our documentation.


Is there a way to make this authentication more durable? 2 weeks seems a bit short for an automated process.


I asked team behind quartopub service about authentication delay when using access token, if any as we interactive credential flow. I'll report what they say when I have an answer.

**Reply by cderv** [reply]
> Thanks - building an export token line into the start of the cron job works very well.Hindsight - Just wish I had done that from the beginning.Peter

Awesome. Glad it works !
